### ISSUE TITLE
Move build tools and utilities into separate repo and publish them as NuGet packages

### ISSUE THREAD

**Issue by dariuszparys** [OpenLiveWriter/OpenLiveWriter#13]
Per @Eilon 's comment

> The way we solve that in ASP.NET is by splitting build tools into a separate process:
> 1. The source code for build tools are typically in a separate repo that is built only when their source code changes.
> 2. When their source code changes, they are compiled and the build outputs are packaged as NuGet packages and uploaded to a feed on MyGet that has all the build tools.
> 3. The main repo (i.e. this repo) uses the NuGet packages to restore the build tools and then run them.
> 
> This avoids the nastiness of checking binaries into a Git repo, and also avoids the hassle/slowness of recompiling the tools on every build.

@willduff

**Comment by rebornix** [comment]
Separate the build and publish utilities would be cool. 

IS it possible taht we make the utilities available in Xamarin platform ? In which case we can easily port OLW to iOS and Android.

**Comment by Eilon** [comment]
@rebornix the app depends heavily on Windows Forms and MSHTML (basically part of IE). Anything outside of Win32 is a big challenge that is probably out of scope for this issue (but others have logged discussion issues to support non-Windows platforms).

**Comment by MitchMilam** [comment]
@rebornix as @Eilon mentioned, quite a bit of work needs to be done before we can port the code to Xamarin. We need to get OLW to run on other desktop platforms before jumping into mobile. Hopefully, Xamarin Forms will keep advancing and that will make the final transition to mobile much less of an issue.

**Comment by shanselman** [comment]
Thanks for reporting this and for helping Open Live Writer over the years. We are doing an issue-bankruptcy cleanup so we can make the repository maintainable again and focus on current, actionable work. This older issue is being closed as part of that cleanup, not because it was unimportant. If this still reproduces in the current code or is still relevant to the revitalization work, please open a fresh issue with current details and we will triage it going forward.
